continuum/metrics/logger.py

```diff
@@ -1,128 +1,80 @@
-
-    def add_step(self, predictions=None, targets=None, task_ids=None, subset="test", model=None):
-        if subset not in ("train", "test"):
-            raise ValueError(f"Subset must be train, val, or test, not {subset}.")
-
-        if isinstance(predictions, torch.Tensor):
-            predictions = predictions.cpu().numpy()
-        if isinstance(targets, torch.Tensor):
-            targets = targets.cpu().numpy()
-        if isinstance(task_ids, torch.Tensor):
-            task_ids = task_ids.cpu().numpy()
-
-        if predictions is not None and targets is not None and task_ids is not None:
-            self._predictions[subset].append(predictions)
-            self._targets[subset].append(targets)
-            self._tasks[subset].append(task_ids)
-
-        if model is not None:
-            self._model_sizes.append(get_model_size(model))
-
-        # Remove all cached properties
-        for k in list(self.__dict__.keys()):
-            if k.startswith("__cached_"):
-                del self.__dict__[k]
-
-        self._batch_predictions = []
-        self._batch_targets = []
+
+class Logger(_BaseLogger):
+    def __init__(self, root_log=None, list_keywords=["performance"], list_subsets=["train", "test"]):
+        super().__init__(root_log=root_log, list_keywords=list_keywords, list_subsets=list_subsets)
 
     def log(self):
         print(f"Task id={self.nb_tasks}, acc={self.accuracy}, avg-acc={self.average_incremental_accuracy}")
 
     @property
     def nb_tasks(self):
-        return len(self._predictions[list(self._predictions.keys())[0]])
+        return self.current_task
+
+    def _conv_list_vector(self, list_vector):
+        if len(list_vector) > 1:
+            vector = np.concatenate(list_vector)
+        else:
+            vector = list_vector[0]
+        return vector
+
+    def get_all_last_epoch_data(self, subset):
+
+        last_epoch_pred = []
+        last_epoch_targets = []
+        last_epoch_task_ids = []
+        for task_id in range(self.current_task):
+            predictions = self.logger_dict["performance"][subset][task_id][-1]["predictions"]
+            targets = self.logger_dict["performance"][subset][task_id][-1]["targets"]
+            task_id = self.logger_dict["performance"][subset][task_id][-1]["task_ids"]
+
+            last_epoch_pred.append(predictions)
+            last_epoch_targets.append(targets)
+            last_epoch_task_ids.append(task_id)
+
+        return last_epoch_pred, last_epoch_targets, last_epoch_task_ids
 
     @property
+    @require_subset("train")
     def online_accuracy(self):
-        if len(self._batch_predictions) == 0:
+        if self._get_current_predictions("train").size == 0:
             raise Exception(
-                "You need to call <add_batch(preds, targets)> in order to get the online accuracy."
+                "You need to call add([prediction, label, task_id]) in order to compute an online accuracy "
+                "(task_id might be None)."
```

Comment:
we never use the task_id but that the classical way to use the `add` function

Reply:
Oh ok I understood, I though the error message meant that an error happened maybe because the task_id was None.

Anyway, we would automatically check that the user provides a list of three items no (x, y, t)? So that the user cannot forget to give a (potentially None) task_id.

Reply:
ah ok, I will modify it.